packages/lib/src/utils/string/string.test.ts

```diff
@@ -1,8 +1,21 @@
-import { toSentenceCase } from ".";
+import { toSentenceCase, toEscapedRegex } from ".";
 
 describe("toSentenceCase", () => {
   it("capitalizes the first letter in a word", () => {
     expect(toSentenceCase("mystring")).toBe("Mystring");
     expect(toSentenceCase("myString")).toBe("Mystring");
   });
 });
+
+describe("toEscapedRegex", () => {
+  it("escapes special characters in a string", () => {
+    // eslint-disable-next-line no-useless-escape -- This is a regex escape character test
+    expect(toEscapedRegex("hello-world")).toBe("hello\\-world");
+    // eslint-disable-next-line no-useless-escape -- This is a regex escape character test
+    expect(toEscapedRegex("hello/world")).toBe("hello\\/world");
+    expect(toEscapedRegex("https://example.com")).toBe(
+      // eslint-disable-next-line no-useless-escape -- This is a regex escape character test
+      `https:\\/\\/example\\\.com`,
+    );
+  });
```

Comment:
I feel like this will pass the tests and the linting:
```suggestion
  it("escapes special characters in a string", () => {
    expect(toEscapedRegex("hello-world")).toBe("hello\\-world");
    expect(toEscapedRegex("hello/world")).toBe("hello\\/world");
    expect(toEscapedRegex("https://example.com")).toBe(
      "https:\\/\\/example\\.com",
    );
  });
```